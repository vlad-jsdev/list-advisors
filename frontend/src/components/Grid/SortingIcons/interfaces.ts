export interface ISortingIcons {
	direction?: 'asc' | 'desc';
}