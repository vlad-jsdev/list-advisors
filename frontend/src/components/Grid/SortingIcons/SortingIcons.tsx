import { FC } from 'react';

import ArrowDownward from '@mui/icons-material/ArrowDownward';
import ArrowUpward from '@mui/icons-material/ArrowUpward';
import { ISortingIcons } from './interfaces';

const SortingIcons: FC<ISortingIcons> = ({ direction }) => (
	direction === 'asc'
		? <ArrowUpward style={{ fontSize: '18px' }} data-testid="upward-icon" />
		: <ArrowDownward style={{ fontSize: '18px' }} data-testid="downward-icon" />
);

export default SortingIcons