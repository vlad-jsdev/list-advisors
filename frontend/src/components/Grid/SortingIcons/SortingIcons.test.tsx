import React from 'react';
import { render, screen } from '@testing-library/react';
import SortingIcons from './SortingIcons';

describe('SortingIcons', () => {
  it('renders the SortingIcons component with the correct direction', () => {
    render(<SortingIcons direction="asc" />);
    const upwardIcon = screen.getByTestId('upward-icon');
    expect(upwardIcon).toBeInTheDocument();

    render(<SortingIcons direction="desc" />);
    const downwardIcon = screen.getByTestId('downward-icon');
    expect(downwardIcon).toBeInTheDocument();
  });
  it('matches the snapshot with the correct direction', () => {
    const { asFragment } = render(<SortingIcons direction="asc" />);
    expect(asFragment()).toMatchSnapshot();

    const { asFragment: asFragment2 } = render(<SortingIcons direction="desc" />);
    expect(asFragment2()).toMatchSnapshot();
  });
});