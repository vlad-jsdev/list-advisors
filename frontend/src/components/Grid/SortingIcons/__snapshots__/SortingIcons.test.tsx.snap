// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingIcons matches the snapshot with the correct direction 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="upward-icon"
    focusable="false"
    style="font-size: 18px;"
    viewBox="0 0 24 24"
  >
    <path
      d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
</DocumentFragment>
`;

exports[`SortingIcons matches the snapshot with the correct direction 2`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="downward-icon"
    focusable="false"
    style="font-size: 18px;"
    viewBox="0 0 24 24"
  >
    <path
      d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
    />
  </svg>
</DocumentFragment>
`;
